pkg/band/band.go

```diff
@@ -300,248 +304,368 @@ var usAuBeaconFrequencies = func() (freqs [8]uint64) {
 	return freqs
 }()
 
+func parseChMask(offset uint8, mask ...bool) map[uint8]bool {
+	if len(mask)-1 > int(math.MaxUint8-offset) {
+		panic(fmt.Sprintf("channel mask overflows uint8, offset: %d, mask length: %d", offset, len(mask)))
+	}
+	m := make(map[uint8]bool, len(mask))
+	for i, v := range mask {
+		m[offset+uint8(i)] = v
+	}
+	return m
+}
+
 func parseChMask16(mask [16]bool, cntl uint8) (map[uint8]bool, error) {
-	chans := make(map[uint8]bool, 16)
 	switch cntl {
 	case 0:
-		for i := uint8(0); i < 16; i++ {
-			chans[i] = mask[i]
-		}
+		return parseChMask(0, mask[:]...), nil
 	case 6:
-		for i := uint8(0); i < 16; i++ {
-			chans[i] = true
-		}
-	default:
-		return nil, errUnsupportedChMaskCntl.WithAttributes("chmaskcntl", cntl)
+		return parseChMask(0,
+			true, true, true, true, true, true, true, true,
```

Comment:
Can we introduce a small `repeat(count, value)` function ? I find it error prone to write the mask by hand. You can even create the result in `init` and reuse it from there, just don't put 16 `true`s one after another.

Reply:
I took this approach, because it's simpler and it's obvious what's going on. I'd argue that constructing these by using utility functions is actually more error-prone, since then you have to merge stuff.